[docs]grpc datasource (#6242)
ToolJet/ToolJet: docs/docs/data-sources/grpc.md

---
id: grpc
title: gRPC
---

<div className='badge badge--primary heading-badge'>Self-hosted only</div>

:::caution

> At present, gRPC can be deployed for self-hosting and has the capability to handle unary requests and responses.

Reviewer: Can we write something like this
```suggestion
only self-hosted deployments will have access to a gRPC datasource that is capable of handling unary requests and responses.
```

:::

- [Setup](#setup)
- [Querying gRPC](#querying-grpc)

## Setup

### Step 1: Upgrade ToolJet to the version 2.5 or above

Find instructions on how to do this in the setup guides located here: https://docs.tooljet.com/docs/setup/.

### Step 2: Add proto files

> Add proto files by creating a directory named "protos" and placing a file called "**service.proto**" inside it.

Reviewer: we should mention the **protos** dir should be in the root

> Add proto files by creating a directory named "protos" and placing a file called "**service.proto**" inside it.

Author: ```suggestion
At the root, create a directory named "**protos**" and add a "**service.proto**" file inside it.
```

:::info Note

> Copy-pase the current service.proto service

Reviewer: We can remove this, not relevant

> :::info Note
> Copy-pase the current service.proto service
> :::

Author: ```suggestion
```

### Step 3: Mount Volumes

In the `docker-compose.yml` add

```bash
./protos:/app/protos
```

to the 2 volume sections for **plugins** and **server**

<div style={{textAlign: 'center'}}>

<img className="screenshot-full" src="/img/datasource-reference/grpc/proto1.png" alt="gRPC: datasource" width='500' />

</div>

<div style={{textAlign: 'center'}}>

<img className="screenshot-full" src="/img/datasource-reference/grpc/proto2.png" alt="gRPC: datasource"  width='500'/>

</div>

### Step 4: Reboot the instance

```bash
docker-compose up -d
```

## Querying gRPC

After setting up your proto files, you should be able to establish a connection to gRPC by going to the [global datasource](/docs/data-sources/overview) page.

### Connect the gRPC datasource

> ToolJet requires the following to connect to Amazon SES:

Reviewer: gRPC servers

> ToolJet requires the following to connect to Amazon SES:

Author: ```suggestion
ToolJet requires the following to connect to gRPC servers:
```

- **Server URL**
- **Authentication type** (None, Basic, Bearer, and API key)

<div style={{textAlign: 'center'}}>

<img className="screenshot-full" src="/img/datasource-reference/grpc/connection.png" alt="gRPC: connection" />

</div>

Once you have added the gRPC from the global datasource page, you'll find it on the query panel of the application.

<div style={{textAlign: 'center'}}>

<img className="screenshot-full" src="/img/datasource-reference/grpc/grpcgds.png" alt="gRPC: connection" />

</div>

### Creating query

> You can now query a particular of RPC of the added services.

Reviewer: ```suggestion
You can now query a particular of RPC method of the added services.
```

> You can now query a particular of RPC of the added services.

Author: ```suggestion
You can now query a particular RPC method of the added services.
```
